src/services/outliningElementsCollector.ts

```diff
@@ -73,9 +73,12 @@ namespace ts.OutliningElementsCollector {
         const lineStarts = sourceFile.getLineStarts();
         for (let i = 0; i < lineStarts.length; i++) {
             const currentLineStart = lineStarts[i];
-            const lineEnd = i + 1 === lineStarts.length ? sourceFile.getEnd() : lineStarts[i + 1] - 1;
+            const newlinePosition = i + 1 === lineStarts.length ? sourceFile.getEnd() : lineStarts[i + 1] - 1;
+            // If the character before \n is \r, decrement line end by 1.
+            const lineEnd = sourceFile.text.charCodeAt(newlinePosition - 1) === CharacterCodes.carriageReturn ? newlinePosition - 1 : newlinePosition;

```

Comment:
Can you use sourceFile.getLineEndOfPosition instead then ? (From https://github.com/microsoft/TypeScript/pull/31220#issuecomment-489712866)

Reply:
That would make sense, thanks!